ccl/revertccl: TestRevertSpansFanout failed

ccl/revertccl.TestRevertSpansFanout [failed](https://mesolite.cluster.engflow.com/invocations/default/3ea6743e-8005-40a7-ac47-aa1b11395f5c?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3JldmVydGNjbDpyZXZlcnRjY2xfdGVzdA==) on release-24.3.25-rc @ [eecd7c2c316ed213daab1c664e0d854fc937a746](https://github.com/cockroachdb/cockroach/commits/eecd7c2c316ed213daab1c664e0d854fc937a746):


```
        	            	  |   | 	pkg/kv/kvclient/kvcoord/transport.go:255
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*grpcTransport).SendNext
        	            	  |   | 	pkg/kv/kvclient/kvcoord/transport.go:183
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.raceTransport.SendNext
        	            	  |   | 	pkg/kv/kvclient/kvcoord/transport_race.go:71
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendToReplicas
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:2748
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendPartialBatch
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:2203
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).divideAndSendBatchToRanges
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:1957
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).Send
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:1257
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*CrossRangeTxnWrapperSender).Send
        	            	  |   | 	pkg/kv/db.go:224
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*DB).sendUsingSender
        	            	  |   | 	pkg/kv/db.go:1133
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*DB).send
        	            	  |   | 	pkg/kv/db.go:1116
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.sendAndFill
        	            	  |   | 	pkg/kv/db.go:965
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*DB).Run
        	            	  |   | 	pkg/kv/db.go:988
        	            	  |   | github.com/cockroachdb/cockroach/pkg/ccl/revertccl.RevertSpans
        	            	  |   | 	pkg/ccl/revertccl/revert.go:139
        	            	  |   | github.com/cockroachdb/cockroach/pkg/ccl/revertccl.RevertSpansFanout.func2
        	            	  |   | 	pkg/ccl/revertccl/revert.go:97
        	            	  |   | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  |   | 	external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  |   | runtime.goexit
        	            	  |   | 	src/runtime/asm_amd64.s:1695
        	            	  | Wraps: (2) failed to connect to n3 at 127.0.0.1:42955
        	            	  | Wraps: (3) forced error mark
        	            	  |   | "originated at breaker breaker"
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/circuit/*circuit.breakerErrorMark::
        	            	  | Wraps: (4) forced error mark
        	            	  |   | "breaker open"
        	            	  |   | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	  | Wraps: (5) grpc: refusing to dial; node is quiescing [code 7/PermissionDenied]
        	            	  | Error types: (1) *errbase.opaqueWrapper (2) *errutil.withPrefix (3) *markers.withMark (4) *markers.withMark (5) *status.Error
        	            	Wraps: (5) node unavailable; try another peer
        	            	Error types: (1) *errbase.opaqueWrapper (2) *errutil.withPrefix (3) *kvpb.AmbiguousResultError (4) *secondary.withSecondaryError (5) *kvpb.NodeUnavailableError
        	Test:       	TestRevertSpansFanout/revert-fanout-with-callback
    --- FAIL: TestRevertSpansFanout/revert-fanout-with-callback (808.78s)
=== RUN   TestRevertSpansFanout/revert-with-1-worker
    revert_test.go:128: error executing query="DELETE FROM test WHERE k % 5 = 2" args=[]: ERROR: dial tcp 127.0.0.1:39275: connect: connection refused
    --- FAIL: TestRevertSpansFanout/revert-with-1-worker (0.08s)
=== RUN   TestRevertSpansFanout/revert-without-callback
    revert_test.go:121: error executing query="DELETE FROM test WHERE k % 5 = 2" args=[]: ERROR: dial tcp 127.0.0.1:39275: connect: connection refused
    --- FAIL: TestRevertSpansFanout/revert-without-callback (0.10s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRevertSpansFanout.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-58718

--- a/pkg/revert/BUILD.bazel
+++ b/pkg/revert/BUILD.bazel
@@ -65,6 +65,7 @@ go_test(
         "//pkg/sql/sem/catid",
         "//pkg/testutils",
         "//pkg/testutils/serverutils",
+        "//pkg/testutils/skip",
         "//pkg/testutils/sqlutils",
         "//pkg/testutils/testcluster",
         "//pkg/util/hlc",

--- a/pkg/revert/revert_test.go
+++ b/pkg/revert/revert_test.go
@@ -23,6 +23,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/catid"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
@@ -38,6 +39,8 @@ import (
 func TestRevertSpansFanout(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+
+	skip.UnderRace(t, "takes too long under race")
 
 	ctx := context.Background()
 

